src/base/net/smtp.cpp

```diff
@@ -188,11 +188,11 @@ void Smtp::readyRead()
     {
         const int pos = m_buffer.indexOf("\r\n");
         if (pos < 0) return; // Loop exit condition
-        const QByteArray line = m_buffer.left(pos);
+        const auto line = QString::fromUtf8(m_buffer.left(pos));
```

Comment:
Finally I looked through all these changes.
This is the only place that is confusing. I wouldn't do that (at least within the scope of this PR).
The received data is used as a "raw" QByteArray everywhere, except for output to the Log, so it is at least illogical to convert it to QString and then back to QByteArray. I would just leave the logic as it is, and convert it to QString only where it is already converted (implicitly).

Reply:
Changed.

Also note that with `QT_NO_CAST_FROM_ASCII` defined, the conversion from QByteArray to QString must be explicit, the implicit conversion is not available.

Reply:
> Also note that with `QT_NO_CAST_FROM_ASCII` defined, the conversion from QByteArray to QString must be explicit, the implicit conversion is not available.

Yes, I know.